src/main/java/dev/openfeature/sdk/FeatureProviderStateManager.java

```diff
@@ -24,43 +24,60 @@ public void initialize(EvaluationContext evaluationContext) throws Exception {
         }
         try {
             delegate.initialize(evaluationContext);
-            state = ProviderState.READY;
+            setState(ProviderState.READY);
         } catch (OpenFeatureError openFeatureError) {
             if (ErrorCode.PROVIDER_FATAL.equals(openFeatureError.getErrorCode())) {
-                state = ProviderState.FATAL;
+                setState(ProviderState.FATAL);
             } else {
-                state = ProviderState.ERROR;
+                setState(ProviderState.ERROR);
             }
             isInitialized.set(false);
             throw openFeatureError;
         } catch (Exception e) {
-            state = ProviderState.ERROR;
+            setState(ProviderState.ERROR);
             isInitialized.set(false);
             throw e;
         }
     }
 
     public void shutdown() {
         delegate.shutdown();
-        state = ProviderState.NOT_READY;
+        setState(ProviderState.NOT_READY);
         isInitialized.set(false);
     }
 
     @Override
     public void onEmit(ProviderEvent event, ProviderEventDetails details) {
         if (ProviderEvent.PROVIDER_ERROR.equals(event)) {
             if (details != null && details.getErrorCode() == ErrorCode.PROVIDER_FATAL) {
-                state = ProviderState.FATAL;
+                setState(ProviderState.FATAL);
             } else {
-                state = ProviderState.ERROR;
+                setState(ProviderState.ERROR);
             }
         } else if (ProviderEvent.PROVIDER_STALE.equals(event)) {
-            state = ProviderState.STALE;
+            setState(ProviderState.STALE);
         } else if (ProviderEvent.PROVIDER_READY.equals(event)) {
-            state = ProviderState.READY;
+            setState(ProviderState.READY);
+        }
+    }
+
+    private void setState(ProviderState state) {
+        ProviderState oldState = this.state.getAndSet(state);
+        if (oldState != state) {
+            String providerName;
+            if (delegate.getMetadata() == null || delegate.getMetadata().getName() == null) {
+                providerName = "unknown";
+            } else {
+                providerName = delegate.getMetadata().getName();
+            }
+            log.info("Provider {} transitioned from state {} to state {}", providerName, oldState, state);
```

Comment:
I think this should be a debug log.

Reply:
I understand why you'd say so, and I think I would agree most times, but the point of this change was to get better visibly into state transitions, particularly for when providers become ready again. Because this isn't in a hot code path, I think it might be worthwhile to have. We've seen internally that such a log would be very useful to confirm the provider is back in a good state.

If we make it debug, we'll never see it "after the fact" if logging it set to normal level, so I think it will make troubleshooting issues harder. I think the chances of spam here are very low in normal situations.

Reply:
I am not sure, we need this log all the time. imho I think it is important when we are going from error back to ready - but the other transitions are not as important to see. Stale can happen, we have a gracePeriod, I do not need to know it. but after an error, I need to know when got back to ready